### PR TITLE
fix for adding parameters with user-defined symbols

### PR DESCRIPTION
**Pull request by newville**
This addresses #607 

###### Type of Changes

- [x ] Bug fix
- [ ] New feature
- [ ] Refactoring / maintenance
- [ ] Documentation / examples

###### Tested on

Tests (including added test based on #607) works for me with

Python: 3.7.4 (default, Aug 13 2019, 20:35:49) 
[GCC 7.3.0]

lmfit: 0.9.14+40.gd001938, scipy: 1.3.1, numpy: 1.17.2, asteval: 0.9.16, uncertainties: 3.0.2, six: 1.12.0

###### Verification 
Have you

- [ ] included docstrings that follow PEP 257?

- [x] referenced existing Issue and/or provided relevant link to mailing list?

- [x] verified that existing tests pass locally?

- [ ] verified that the documentation builds locally?

- [x] squashed/minimized your commits and written descriptive commit messages?

- [ ] added or updated existing tests to cover the changes?
- [ ] updated the documentation?
- [ ] added an example?

### CODE REVIEW

File: lmfit/parameter.py [hunk 1]
```diff
@@ -75,12 +75,19 @@ def __init__(self, asteval=None, usersyms=None, *args, **kwds):
...
             self._asteval.symtable[key] = val
 
-        self.update(*args, **kwds)
-
     def copy(self):
         """Parameters.copy() should always be a deepcopy."""
         return self.__deepcopy__(None)
 
+    def update(self, other):
+        """update values with another Paramters object"""
```
Comment (reneeotten) [inline lmfit/parameter.py]: how about ```Update values and symbols with another Parameters object.``` - correcting a typo, adding "symbols" and using uppercase/punctuation.

File: tests/test_parameters.py [hunk 2]
```diff
@@ -299,3 +300,24 @@ def test_params_prints(self):
         assert len(repr_full) > 150
         assert len(repr_one) > 150
         assert len(out) > 150
+
+    def test_add_with_symtabale(self):
```
Comment (reneeotten) [inline tests/test_parameters.py]: ```symtabale``` --> ```symtable```  and can we add as docstring ```"""Regression test for ﻿GitHub Issue 607."""```

**Comment by newville** [comment]
@reneeotten  Thanks -- I addressed typos and suggestions.  Yeah, the whole test suite is looking a bit old and unorganized...

**Comment by reneeotten** [comment]
> @reneeotten Thanks -- I addressed typos and suggestions. 

You don't like the addition of a docstring for the test, so that it's clear where this came from, and a ```.```  at the end of a docstring ;)

> Yeah, the whole test suite is looking a bit old and unorganized...

Yes, I will continue on that, but it's quite tedious and not the most fun part...

**Comment by newville** [comment]
@reneeotten  Most of the lmfit tests have a comment or two, but no docstrings.  These aren't public methods.    Personally, I would not worry too much about re-organizing the tests.  

I will fix the period at the end of the docstring, and then I will merge this.

**Pull request merged**